src/lib/components/Error/ErrorReporter.tsx

```diff
@@ -0,0 +1,23 @@
+import { useAtomValue } from 'jotai/utils'
+import { providerAtom } from 'lib/state/web3'
+import { useEffect } from 'react'
+import { EMPTY } from 'widgets-web3-react/empty'
+
+class IntegrationError extends Error {
+  constructor(message: string) {
+    super(message)
+    this.name = 'INTEGRATION ERROR'
+  }
+}
+
+const missingProviderError = new IntegrationError('Missing provider')
```

Comment:
better to define this inline for readability - if we're throwing an error, whether it's a singleton doesn't matter as much.